results-processor/wptreport.py

```diff
@@ -523,20 +543,22 @@ def prepare_labels(report: WPTReport,
     Returns:
         A set of strings.
     """
+    browser = report.run_info['product']
+    # browser_channel is an optional field.
+    channel = report.run_info.get('browser_channel')
     labels = set()
-    labels.add(report.run_info['product'])
+    labels.add(browser)
     labels.add(uploader)
     # Empty labels may be generated here, but they will be removed later.
     for label in labels_str.split(','):
         labels.add(label.strip())
 
     # Add the release channel label.
-    if report.run_info.get('browser_channel'):
-        labels.add(report.run_info['browser_channel'])
-        if report.run_info['browser_channel'] in CHANNEL_TO_LABEL:
-            labels.add(CHANNEL_TO_LABEL[report.run_info['browser_channel']])
-    elif not any([i in labels for i in set(CHANNEL_TO_LABEL.values())]):
-        # Default to "stable".
+    if channel:
+        labels |= _channel_to_labels(browser, channel)
+    elif not (labels & RELEASE_CHANNEL_LABELS):
+        # Default to "stable" if no channel label or browser_channel is present
+        # TODO(Hexcles): remove this fallback default eventually.
```

Comment:
Do we have a way to track that this branch is happening? How will we know when it is safe to remove?

Reply:
Great point! I've added some logging to see if this still happens.